pdf-service/src/index.js

```diff
@@ -500,6 +502,64 @@ app.post(
   })
 );
 
+app.post(
+  "/pdf-service/v1/_getUnrigesteredCodes",
+  asyncHandler(async (req, res) => {
+    let requestInfo;
+    try {
+      requestInfo = get(req.body, "RequestInfo");
+      res.status(200);
+      res.json({
+          ResponseInfo: requestInfo,
+          unregisteredLocalisationCodes: unregisteredLocalisationCodes,
+        });
+    } catch (error) {
+      logger.error(error.stack || error);
+      res.status(500);
+      res.json({
+        ResponseInfo: requestInfo,
+        message: "Error while retreving the codes",
+      });
+    }
+  })
+
+);
+
+app.post(
+  "/pdf-service/v1/_clearUnrigesteredCodes",
+  asyncHandler(async (req, res) => {
+    let requestInfo;
+    try {
+      requestInfo = get(req.body, "RequestInfo");
+      let resposnseMap = await findLocalisation(
+        requestInfo,
+        [],
+        unregisteredLocalisationCodes
+      );
+
+      resposnseMap.messages.map((item) => {
+        if(unregisteredLocalisationCodes.includes(item.code)){
+          var index = unregisteredLocalisationCodes.indexOf(item.code);
+          unregisteredLocalisationCodes.splice(index, 1);
+        }
+      });
+      res.status(200);
+      res.json({
+          ResponseInfo: requestInfo,
+          unregisteredLocalisationCodes: unregisteredLocalisationCodes,
+        });
+    } catch (error) {
+      logger.error(error.stack || error);
+      res.status(500);
```

Comment:
We should never throw any `500` errors, all errors should also be `400` and use the format that we use in other services. Check the whole service for fixes

Reply:
updated error status

Reply:
updated error status